### DISCUSSION TITLE
Feature Request - Display Multiple sensors on one button

### DISCUSSION THREAD

**Discussion by cedi** [cgiesche/streamdeck-homeassistant#204] [Feature Requests]
Hi there,
thank you for your incredible work.

I am primarily using this Plugin to display sensor states from my home-office on my streamdeck.

What I absolutely love the this plugin so far is the fact that I can display a sensor value and can call different actions with the short and long press buttons, all on the same streamdeck key.

However I would love the possibility to display more than just one sensor value (entity) per key.

In my specific example, I have a temp, TVOC, CO2, and PM2.5 sensor in my home-office.
I also have an air-purifier and HVAC, both integrated in homeassistant.
I would love to display both the TVOC and PM2.5 values on a button where I can then toggle my air-purifier. And on a second button I'd love to display my Temp and CO2 and toggle the HVAC.

One possible implementation for this could be by leveraging the existing "Custom labels" section, or adding a new "Advanced labels" section to display multiple sensors value.
For example In my current setup I use it like this:
![image](https://github.com/cgiesche/streamdeck-homeassistant/assets/1952599/7085bbee-a7f8-4e3e-ab5c-60c47f52cc09)

But I could imagine a configuration that looks like this:
![image](https://github.com/cgiesche/streamdeck-homeassistant/assets/1952599/aace52cb-b5ab-4a5c-bbe6-3ed4832bce91)

Which would look like this on the button:
![image](https://github.com/cgiesche/streamdeck-homeassistant/assets/1952599/e19bf338-9866-4172-b1fe-1153d995c2a6)

What do you think about this?
Would you be open to a PR adding this or should I fork?

Thanks a mil in advance

**Comment by cgiesche** [comment]
Hi cedi,

that sounds like a nice feature. I thought of something similar, but never managed to implement it. Of course, i would very much appreciate your help and a pull request.

Thanks in advance :)

**Comment by JohnChristmann** [comment]
I'd like to see this as well.  The thermohygrometers I use register temperature and humidity readings as separate HA entities, so they have to be displayed on separate buttons currently.  I can't display all the information without running out of space on my Stream Deck.

**Comment by cgiesche** [comment]
This is harder to be implemented, than I first thought. As a workaround, you can create a template entity in home-assistant that contains states of other entities as attributes (like the weather entity) and use that template entity on a button.

**Comment by exSnake** [comment]
I would like it too for example manage 2 togglable light with one button, one with long press and one with short press and so you can show
Light1 ON
Light2 OFF in the ltable

I don't think it's possible with template entity
